api/proxy/proxy.go

```diff
@@ -326,12 +328,13 @@ func (p *proxy) paranoiaDAVcheck(normalizedPath string, w http.ResponseWriter, r
 	// check if the url contains remote.php/webdav in another location of the url
 	// the remote.php/webdav endpoint used by the web UI is whitelisted.
 
-	if !strings.HasPrefix(normalizedPath, "/remote.php/webdav") {
-		if strings.Contains(normalizedPath, "remote.php/webdav") {
-			msg := fmt.Sprintf("CRITICAL: webdav path not controlled in logic rules => %s", normalizedPath)
-			p.logger.Error(msg, zap.String("normalizedPath", normalizedPath))
-			panic("CRITICAL: " + normalizedPath)
-		}
+	if strings.HasPrefix(normalizedPath, "/remote.php/webdav") ||
```

Comment:
This panic has different logic than before. We were checking for `/cernbox/remote.php/...` and `/mobile/remote.php/...`, but now we are not checking for those at all. 
On the contrary, we're checking for dav requests that should always land on web, since "/remote.php" is a known web path, so this if has no effect.
I wonder if we should just panic in any condition (i.e remove the if), since we should never reach this point.

Reply:
I missed the point where the code was checking with `strings.Contains` rather than `strings.HasPrefix`.
I have reverted to original code and added extra condition for the new dav routes.
I have to re-order the paranoia block and send it one section down because if not would have panicked as there is this route `cernbox/mobile/remote.php/dav/files` that is sent to the old PHP server and would have been catched by the panic code.